TinyMCE.Blazor/Editor.razor

```diff
@@ -18,186 +20,307 @@
 
 @code {
 
-  private string _value = "";
-  private string _text = "";
-  private bool _setDisable = false;
-  private bool _disable = false;
-  private Dictionary<string, string> streamMap = new Dictionary<string, string>();
-  private int _chunkSize = 30000;
-  private bool _initialized = false;
-  private ElementReference Element;
-  private Dictionary<string, object> blazorConf;
-  private DotNetObjectReference<Editor> dotNetRef;
-
-  [Parameter] public string ClassName { get; set; } = "tinymce-wrapper";
-  [Parameter] public string Id { get; set; } = System.Guid.NewGuid().ToString();
-  [Parameter] public bool Inline { get; set; } = false;
-  [Parameter] public string ApiKey { get; set; } = "no-api-key";
-  [Parameter] public string CloudChannel { get; set; } = "5";
-  [Parameter] public string ScriptSrc { get; set; }
-  [Parameter] public string InitialValue { get; set; }
-  [Parameter] public string JsConfSrc { get; set; }
-  [Parameter] public string ModelEvents { get; set; } = "change input undo redo";
-  [Parameter] public Dictionary<string, object> Conf { get; set; } = new Dictionary<string, object>();
-  [Parameter(CaptureUnmatchedValues = true)] public Dictionary<string, object> AdditionalAttributes { get; set; } = new Dictionary<string, object>();
-
-  [Parameter]
-  public string Value
-  {
-    get => _value;
-    set
-    {
-      string newVal = value == null ? "" : value;
-      if (newVal != _value)
-      {
-        _value = newVal;
-        _ = PushValue(newVal);
-      }
+    private string _value = "";
+    private string _text = "";
+    private bool _setDisable = false;
+    private bool _disable = false;
+    private Dictionary<string, string> streamMap = new Dictionary<string, string>();
+    private int _chunkSize = 30000;
+    private bool _initialized = false;
+    private ElementReference Element;
+    private Dictionary<string, object> blazorConf;
+    private DotNetObjectReference<Editor> dotNetRef;
+    private string _userClassName = "tinymce-wrapper";
+
+    [CascadingParameter]
+    public EditContext EditContext { get; set; }
```

Comment:
I feel conflicted about this since having `CascadingParameter -> EditContext` means the editor can only be used inside an EditForm. Would people use the editor outside an EditForm? If we are bound to be inside an EditForm, would it make more sense to inherit from `InputBase` and let it handle the css and validation? We could get rid of the `For` property in that case

Reply:
When people use the editor outside of an EditForm, the CascadingParameter EditContext would be null and everything should work as it was. I have considered it and that is why the Editor would throw an error to ask people to add the `For` parameter for validation only when the Editor has a valid EditContext.